### ISSUE TITLE
Initialization fails with async tree content.

### ISSUE THREAD

**Issue by dandee** [valor-software/ng2-tree#35]
With asynchronous tree content population, [tree] bound field is initially uninitialized before data comes back from backend. This causes exception in TreeInternalComponent.ngOnInit:

```
core.umd.js:3472 EXCEPTION: Uncaught (in promise): Error: Error in ./TreeComponent class TreeComponent - inline template:0:15 caused by: Cannot set property '_indexInParent' of undefined
TypeError: Cannot set property '_indexInParent' of undefined
    at TreeInternalComponent.ngOnInit (http://localhost:8081/node_modules/ng2-tree/bundles/ng2-tree.umd.js:5883:38)
```

Current workaround is to initialize bound field with some dummy `TreeModel` object - but it's ugly (well, it can be used as poor mans "Loading" indicator but you can better than that ;) ).
The latter reminds me that it'd be also great to have a initialization substitute content (like <template>) for the tree.

To replicate the issue use something like this:
```
// 1 - import required classes and interfaces
import {TreeModel} from 'ng2-tree';

@Component({
  selector: 'myComp',
  // 2 - set [tree] attribute to tree object
  template: `<tree [tree]="tree"></tree>`
})
class MyComponent {
  // 3 - make sure tree object confirms to TreeModel interface
  public tree: TreeModel;

  constructor(private someService: SomeService) { }

  ngOnInit(): void {
    this.someService.getAsyncData().then(tree => this.tree = tree)
  }
}
```

**Comment by rychkog** [comment]
@dandee thanks! Currently I am working on tests - cause it is hard to make changes without them - once finished - will fix this :)

**Comment by rychkog** [comment]
@dandee It is fixed in version `ng2-tree@2.0.0-alpha.3`
